app/addons/fauxton/components.react.jsx

```diff
@@ -334,13 +354,246 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
+    },
+
+    getStoreState: function () {
+      return {
+        isVisible: notificationStore.isNotificationCenterVisible(),
+        filter: notificationStore.getNotificationFilter(),
+        notifications: notificationStore.getNotifications()
+      };
+    },
+
+    componentDidMount: function () {
+      notificationStore.on('change', this.onChange, this);
+    },
+
+    componentWillUnmount: function () {
+      notificationStore.off('change', this.onChange);
+    },
+
+    onChange: function () {
+      if (this.isMounted()) {
+        this.setState(this.getStoreState());
+      }
+    },
+
+    getNotifications: function () {
+      if (!this.state.notifications.length) {
+        return (
+          <li className="no-notifications">No notifications.</li>
+        );
+      }
+
+      return _.map(this.state.notifications, function (notification, i) {
+        return (
+          <NotificationRow
+            isVisible={this.state.isVisible}
+            item={notification}
+            filter={this.state.filter}
+            key={notification.notificationId}
+          />
+        );
+      }, this);
+    },
+
+    selectFilter: function (e) {
+      var filter = $(e.target).closest('li').data('filter');
+      Actions.selectNotificationFilter(filter);
+    },
+
+    render: function () {
+      var panelClasses = 'notification-center-panel flex-layout flex-col';
+      if (this.state.isVisible) {
+        panelClasses += ' visible';
+      }
+
+      var filterClasses = {
+        all: 'flex-body',
+        success: 'flex-body',
+        error: 'flex-body',
+        info: 'flex-body'
+      };
+      filterClasses[this.state.filter] += ' selected';
+
+      var maskClasses = 'notification-page-mask' + ((this.state.isVisible) ? ' visible' : '');
+      return (
+        <div>
+          <div className={panelClasses}>
+
+            <header className="flex-layout flex-row">
+              <span className="fonticon fonticon-bell"></span>
+              <h1 className="flex-body">Notifications</h1>
+              <button type="button" aria-hidden="true" onClick={Actions.hideNotificationCenter}>×</button>
+            </header>
+
+            <ul className="notification-filter flex-layout flex-row" onClick={this.selectFilter}>
+              <li className={filterClasses.all} data-filter="all" title="All notifications">All</li>
+              <li className={filterClasses.success} data-filter="success" title="Success notifications">
+                <span className="fonticon fonticon-ok-circled"></span>
+              </li>
+              <li className={filterClasses.error} data-filter="error" title="Error notifications">
+                <span className="fonticon fonticon-attention-circled"></span>
+              </li>
+              <li className={filterClasses.info} data-filter="info" title="Info notifications">
+                <span className="fonticon fonticon-info-circled"></span>
+              </li>
+            </ul>
```

Comment:
Yeah, this would work... I think whether it's an improvement is debatable, though, having to add 3X the number of event handlers. (What if there were 10 items & they couldn't be added via a loop?) But fair enough point about generally wanting to avoid accessing DOM directly.